Cherry-pick: Allow the launcher command to be configurable (#11888)

mprahl opened the pull request:
**Description of your changes:**

This also updates to UBI 9.

This allows the launcher command to be overridden with the V2_LAUNCHER_COMMAND environment variable. This is useful if you need to override the command to launch Delve for debugging or you have a situation that requires using a different binary in the container image based on the environment.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
(cherry picked from commit 70d28885f27bff40397f9c9ea790e5985deb91e1)

**Checklist:**
- [x] You have [signed off your commits](https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits)
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
  - The command used to launch certain containers can now be configured via an environment variable, offering greater flexibility for deployment environments.

- **Chores**
  - Improved internal handling of container launch commands to support dynamic configuration.
  - Updated base images and build processes for launcher and driver components to newer versions.
  - Removed the backend unit and integration test GitHub Actions workflow.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mprahl committed:
feat(backend): Allow the launcher command to be configurable (#11888)

This allows the launcher command to be overridden with
the V2_LAUNCHER_COMMAND environment variable. This is useful if you need
to override the command to launch Delve for debugging or you have a
situation that requires using a different binary in the container image
based on the environment.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
(cherry picked from commit 70d28885f27bff40397f9c9ea790e5985deb91e1)

dsp-developers commented:
A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-158`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-158`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-158`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-158`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-158`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-158`

dsp-developers commented:
An OCP cluster where you are logged in as cluster admin is required.

The Data Science Pipelines team recommends testing this using the Data Science Pipelines Operator.  Check [here](https://github.com/opendatahub-io/data-science-pipelines-operator) for more information on using the DSPO.

To use and deploy a DSP stack with these images (assuming the DSPO is deployed), first save the following YAML to a file named `dspa.pr-158.yaml`:
```yaml
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: pr-158
spec:
  dspVersion: v2
  apiServer:
    image: "quay.io/opendatahub/ds-pipelines-api-server:pr-158"
    argoDriverImage: "quay.io/opendatahub/ds-pipelines-driver:pr-158"
    argoLauncherImage: "quay.io/opendatahub/ds-pipelines-launcher:pr-158"
  persistenceAgent:
    image: "quay.io/opendatahub/ds-pipelines-persistenceagent:pr-158"
  scheduledWorkflow:
    image: "quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-158"
  mlmd:  
    deploy: true  # Optional component
    grpc:
      image: "quay.io/opendatahub/mlmd-grpc-server:latest"
    envoy:
      image: "registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2"
  mlpipelineUI:
    deploy: true  # Optional component 
    image: "quay.io/opendatahub/ds-pipelines-frontend:pr-158"
  objectStorage:
    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'
```

Then run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines.git
cd data-science-pipelines/
git fetch origin pull/158/head
git checkout -b pullrequest 6cd6f8151a761dbf57befcc9a6ffd3ca8a4a2c98
oc apply -f dspa.pr-158.yaml
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

mprahl committed:
Update to UBI 9 and include FIPS binaries for launcher

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

dsp-developers commented:
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-158`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-158`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-158`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-158`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-158`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-158`

mprahl committed:
Remove the failing Tekton CI

Tekton is not supported in ODH/RHOAI and it will be fully removed as
part of the upcoming KFP 2.5 rebase.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

dsp-developers commented:
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-158`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-158`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-158`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-158`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-158`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-158`

hbelmiro reviewed (commented):
/lgtm

gmfrasca reviewed (commented):
/lgtm

HumairAK commented:
/approve
